File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_sero = 0
    max_garo = 0
    for i in sizes:
        if (i[0]<i[1]):
            i[0], i[1] = i[1], i[0]
        if max_garo < i[0]:
            max_garo = i[0]
        if max_sero < i[1]:
            max_sero = i[1]
    
    return max_garo*max_sero

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    vector<int> garos;
    vector<int> seros;
    // 한 쪽으로 몰기
    for(int i=0; i<sizes.size(); i++) {
        int garo = sizes[i][0];
        int sero = sizes[i][1];
        if (sizes[i][0] >= sizes[i][1]) {
            garos.push_back(sizes[i][0]);
            seros.push_back(sizes[i][1]);
        } else {
            garos.push_back(sizes[i][1]);
            seros.push_back(sizes[i][0]);
        }
    }
    sort(garos.rbegin(), garos.rend());
    sort(seros.rbegin(), seros.rend());
    // cout << garos[0] << " " << seros[0] << "\n";
    answer = garos[0]*seros[0];
    return answer;
}